web/src/main/webapp/my-app/layout/services.js

```diff
@@ -380,8 +384,8 @@ define(['angular', 'jquery'], function(angular, $) {
               }
               result.layout = portlets;
             }
-          } else if ($.isArray(data.layout)) { // layoutDoc
-            result.layout = data.layout;
+          } else if ($.isArray(filteredDataLayout)) { // layoutDoc
```

Comment:
The if (line 353) / else if  (365) / else if (387) is supporting 3 different layout APIs. Three different representations of layout. Only the last is a straight up array of fnames. So the filtering the array of fnames to remove duplicates should probably be inside this third branch of the if rather than before the if. That is, `data.layout` is not reliably an array of fnames until inside this else.

Alternatively, we could get confident that data.layout really is always an array of fnames, simplify away the if.